server/map_rpc_server.go

```diff
@@ -321,6 +306,57 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 	return &trillian.SetMapLeavesResponse{MapRoot: newRoot}, nil
 }
 
+func doPreload(ctx context.Context, tx storage.MapTreeTX, hkv []merkle.HashKeyValue) error {
+	ctx, span := spanFor(ctx, "doPreload")
+	defer span.End()
+
+	readRev, err := tx.ReadRevision(ctx)
+	if err != nil {
+		return err
+	}
+	nidSet := make(map[string]bool)
+	type nodeAndID struct {
+		id   string
+		node storage.NodeID
+	}
+	c := make(chan nodeAndID, 2048)
+	var wg sync.WaitGroup
+
+	for _, i := range hkv {
+		wg.Add(1)
+		go func(k []byte) {
+			nid := storage.NewNodeIDFromHash(k)
```

Comment:
does go let you make this a pointer on this line? Then you could probably leave out the line below.

Reply:
Sadly not :/